src/components/buttons/button-link-edit-browse.jsx

```diff
@@ -64,16 +71,14 @@ class ButtonLinkEditBrowse extends React.Component {
 	 * @method _browseClick
 	 */
 	_browseClick = () => {
-		const instance = this;
-
 		const editor = this.props.editor.get('nativeEditor');
-
 		const url = editor.config.documentBrowseLinkUrl;
+		const cb = editor.config.documentBrowseLinkCallback;
+		const linkTarget = this.linkEditButtonRef.current.state.linkTarget;
 
-		const linkTarget = this.refs.linkEditButton.state.linkTarget;
-
-		// TODO: This should invoke callback or emit an event
-		//       Let's talk about the solution we prefer.
+		if (Lang.isFunction(cb)) {
+			cb.apply(null, [editor, url, linkTarget]);
```

Comment:
The idea of forcing `this` to `null` in the callback, is that to prevent accidental interaction with `window`?

Reply:
yes, exactly